### DISCUSSION TITLE
DRILL_TO_DETAIL - Error: Access is Denied

### DISCUSSION THREAD

**Discussion by srinisubramanian** [apache/superset#25388] [Q&A / Help] [answered]
I use a customized version of Alpha Role to expose dashboards to users in a read only fashion.

After enabling DRILL_TO_DETAIL Feature flag and explore on superset permission, the menu is displayed.  However when I right click and choose the Drill to detail option, then the screen is displayed with an Access is denied message.

What are the permissions needed to be assigned to user for Drill to detail to work?

<img width="948" alt="image" src="https://github.com/apache/superset/assets/8343179/6d719839-424c-4120-8bf7-2bd5e8c96622">

**Comment by srinisubramanian** [comment, marked as answer]
Figured this out:  Need to add permissions for can_samples on datasource to get this to work.

**Reply by MicroUiChandan** [reply]
We need to add this in "Public" role
